#Module2 #Task1 #Task2

TheElidanXD opened the pull request:
Add Module2 homework

Commit:
#Module2 #Task1 #Task2
Add Module2 homework

evgenDer commented on module2/src/user-service.ts:
```diff
@@ -0,0 +1,109 @@
+            abortEarly: false,
+            allowUnknown: false
+        });
+        if (error && error.isJoi) {
```
error?.isJoi

TheElidanXD replied:
Done

evgenDer commented on module2/src/user-service.ts:
```diff
@@ -0,0 +1,109 @@
+        res.locals.user = currentUser;
+        next();
+    } else {
+        res.status(404).json({ message: `User with id ${req.params.id} not found` });
```
req.params.id change to id

TheElidanXD replied:
Done

evgenDer commented on module2/src/user-service.ts:
```diff
@@ -0,0 +1,109 @@
+    res.json(userList);
+});
+
+router.get('/user/:id', (req, res) => {
```
create userRouter and move all requests with user

TheElidanXD replied:
Done in module3 HW

evgenDer commented on module2/src/user-service.ts:
```diff
@@ -0,0 +1,109 @@
+
+router.post('/', validateUserSchema(userSchema), (req, res) => {
+    const newUser: User = {
+        id: uuid(),
```
I think it is better to create a model that should simulate working with database

TheElidanXD replied:
done in Module 3 HW

evgenDer commented on module2/src/user-service.ts:
```diff
@@ -0,0 +1,109 @@
+    if (Number(req.params.limit) < 1) {
+        res.status(400).json({ message: 'Invalid limit' });
+    } else {
+        const autoSuggestUsers = getUndeletedUsers().
```
you shouldn't do it in a router

TheElidanXD replied:
Logic moved to findUsersBySubstring function

evgenDer commented on module2/src/user-service.ts:
```diff
@@ -0,0 +1,109 @@
+const userService = express();
+const router = express.Router();
+
+const userSchema = Joi.object().keys({
```
split all to different files - validation. controllers, routers, and e.t.c

TheElidanXD replied:
Done in Module 3 HW

Commit:
#Module2
Fix check in validation.
Create function to find users by substring.
Fix error message for getting user by id.
